pkg/api/v20240812preview/openshiftcluster_validatestatic.go

```diff
@@ -446,6 +446,18 @@ func (sv openShiftClusterStaticValidator) validateDelta(oc, current *OpenShiftCl
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePropertyChangeNotAllowed, err.Target, err.Message)
 	}
 
+	if current.UsesWorkloadIdentity() {
+		for i := range current.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
```

Comment:
If I'm understanding this correctly, this code will reject an update that changes the order of platform workload identities in the list but otherwise keeps the identity order the same. 

I think we might want to permit such modifications - while the [CLI implementation](https://github.com/Azure/ARO-RP/blob/880835249791f94060b467e5ab58467a8deaecf0/python/az/aro/azext_aro/custom.py#L482) should preserve the existing order, users using ARM/Bicep or Terraform might not guarantee the order of identities here, and as far as I am aware the order of this list should not matter.

Reply:
It's slightly concerning that the order of an ordered list isn't reliable but I've updated the logic to look for a matching `OperatorName` or `ResourceID` in the update data and if found, see if the identity's `ResourceID` or `OperatorName` (respectively) is different

Reply:
I agree with Tanmay, there could be situations where an ARM call has the identities in a different order than the current order in the cluster doc, which would still be valid. thanks @yithian for making it order-agnostic.

Reply:
I don't think the fix you pushed will account for a situation where a cx tries to change the `ResourceID` and the `OperatorName` in a single API call. Here's my idea for how to implement the validation to properly account for this situation:

1. Construct two maps where the keys are operator names and the values are PWI structs - one map for the old cluster doc and one map for the new cluster doc
2. For all keys in new doc that are also in old doc, ensure that all four fields in the structs are equal

I _think_ that would do it, but additional unit test cases would verify for us.